pkg/cnispawn/spawn.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2016 Kinvolk GmbH
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cnispawn
+
+import (
+	"os"
+	"os/exec"
+	"runtime"
+	"syscall"
+)
+
+func RunContainer(path string, background bool) error {
+	runtime.LockOSThread()
+
+	cniNetns, err := NewCniNetns()
+	if err != nil {
+		return err
+	}
+
+	if err := cniNetns.Set(); err != nil {
+		return err
+	}
+	defer cniNetns.Close()
+
+	systemdNspawnPath, err := exec.LookPath("systemd-nspawn")
+	if err != nil {
+		return err
+	}
+
+	// TODO(nhlfr): Don't hardcode kubeadm-systemd specific options,
+	// expose all the options of systemd-nspawn instead.
+	args := []string{
+		systemdNspawnPath,
+		"--capability=cap_audit_write,cap_audit_control,cap_sys_admin,cap_sys_chroot,cap_net_admin,cap_setfcap",
+		"--bind=/sys/fs/cgroup",
+		"-bD",
+		path,
+	}
+
+	if background {
+		_, err := syscall.ForkExec(systemdNspawnPath, args, &syscall.ProcAttr{
```

Comment:
It wouldn't. Not giving stdin to the nspawn process does the trick. And the nspawn process gets owned by systemd (from what I see in `pstree`).